AdyenEncryption/JOSE/Encryption Algorithms/CommonCryptoHelpers.swift

```diff
@@ -0,0 +1,123 @@
+    }
+    let keyData = keyData as NSData
+    let dataIn = dataIn as NSData
+    let initializationVector = initializationVector as NSData?
+    var numBytesOut: size_t = 0
+    let algorithm = CCAlgorithm(kCCAlgorithmAES)
+    let status: CCCryptorStatus = CCCrypt(operation,
+                                          algorithm,
+                                          options,
+                                          keyData.bytes, keyData.length,
+                                          initializationVector?.bytes,
+                                          dataIn.bytes, dataIn.length,
+                                          dataOut.mutableBytes, dataOut.length,
+                                          &numBytesOut)
+    guard status == kCCSuccess else {
+        throw JsonWebEncryptionError.encryptionFailed
+    }
+    let startIndex = dataOut.startIndex
+    let endIndex = startIndex.advanced(by: numBytesOut)
+    return (dataOut as Data).subdata(in: 0..<endIndex)
+}
+
+internal func hmac(data: Data, withKey: Data) throws -> Data {
+    guard let dataOut = NSMutableData(length: Int(CC_SHA512_DIGEST_LENGTH)) else {
+        throw JsonWebEncryptionError.unknownError
+    }
+    let dataIn = data as NSData
+    let keyData = withKey as NSData
+    CCHmac(CCHmacAlgorithm(kCCHmacAlgSHA512),
+           keyData.bytes,
+           keyData.length,
+           dataIn.bytes,
+           dataIn.length,
+           dataOut.mutableBytes)
+    return dataOut as Data
+}
+
+internal func generateRandomData(length: Int) throws -> Data {
+    var bytes = [Int8](repeating: 0, count: length)
+    let status = SecRandomCopyBytes(kSecRandomDefault, length, &bytes)
+    guard status == errSecSuccess else { throw JsonWebEncryptionError.failedToGenerateRandomData }
+    return Data(bytes: bytes, count: length)
+}
+
+internal func secKey(fromModulus modulus: String, exponent: String) throws -> SecKey {
+    guard let modulusHex = modulus.hexadecimal,
+          let exponentHex = exponent.hexadecimal else { throw JsonWebEncryptionError.invalidKey }
+    let keyData = generateRSAPublicKey(with: modulusHex, exponent: exponentHex)
+    var error: Unmanaged<CFError>?
+    let parsedKey = SecKeyCreateWithData(keyData as NSData,
+                                         [
+                                             kSecAttrKeyType: kSecAttrKeyTypeRSA,
+                                             kSecAttrKeyClass: kSecAttrKeyClassPublic
+                                         ] as NSDictionary,
+                                         &error)
+    if let error = error {
+        throw JsonWebEncryptionError.other(error.takeRetainedValue())
+    }
+    
+    guard let key = parsedKey else { throw JsonWebEncryptionError.invalidKey }
+    return key
+}
+
+/// https://github.com/henrinormak/Heimdall/blob/master/Heimdall/Heimdall.swift
+internal func generateRSAPublicKey(with modulus: Data, exponent: Data) -> Data {
```

Comment:
this is copy of [RSA.generateRSAPublicKey](https://github.com/Adyen/adyen-ios/blob/93389d4724ce5d6ed3c75c59b386611e7c1ce86c/AdyenEncryption/Cryptor/RSA.swift#L56)

Reply:
yes, I know, hopefully we'll delete all that old code completely.